### PR TITLE
MAID-1448 Added HolePunchingState

### PR DESCRIPTION
**Pull request by mabeebam**
Hi @benjaminbollen 

Here is my first implementation!

Any [derive] intended for HolePunchingState?
Any impl?
What about variant Mapped?
Type of secret and number_of_failed_attempts correct?

Looking forward to your feedback!

Cheers!

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/749)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @brian-js (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

File: src/lib.rs [hunk 1]
```diff
@@ -108,3 +108,6 @@ pub use name_type::{NameType, closer_to_target, NAME_TYPE_LEN};
 pub use messages::{SignedToken, ExternalRequest, ExternalResponse};
 /// Persona types recognised by the network.
 pub use authority::Authority;
+
+/// Udp Hole Punching
+pub mod connection_management;
```
Comment (benjaminbollen) [inline src/lib.rs]: we don't need this to be public in lib.rs, cheers

**Comment by mabeebam** [comment]
address = 1DddEJsrqR43ajq19nNMUHeTZQ2Biqckr1

**Comment by brian-js** [comment]
Cheers @benjaminbollen, and of course @mabeebam.

**Comment by mabeebam** [comment]
@brian-js My pleasure!

**Comment by NickLambert** [comment]
Thanks @mabeebam, payment of 0.1475 btc on the way

**Pull request merged**